src/duct/__main__.py

```diff
@@ -91,6 +91,71 @@ def max(self, other: ProcessStats) -> ProcessStats:
         )
 
 
+@dataclass
+class Suffix:
+    stdout: str = "stdout"
+    stderr: str = "stderr"
+    usage: str = "usage.json"
+    info: str = "info.json"
+
+    def __iter__(self) -> Generator:
+        for each in fields(self):
+            yield each.name, getattr(self, each.name)
+
+
+@dataclass
+class LogPaths:
+    stdout: str
+    stderr: str
+    usage: str
+    info: str
+
+    def __iter__(self) -> Generator:
+        for each in fields(self):
+            yield each.name, getattr(self, each.name)
+
+    def __post_init__(self) -> None:
+        self._prefix = ""
+
+    @property
+    def prefix(self) -> str:
+        return self._prefix  # type: ignore
+
+    @classmethod
+    def create(cls, output_prefix: str, pid: None | int = None) -> LogPaths:
+        datetime_filesafe = datetime.now().strftime("%Y.%m.%dT%H.%M.%S")
+        formatted_prefix = output_prefix.format(
+            pid=pid, datetime_filesafe=datetime_filesafe
+        )
+        args_dict = {name: f"{formatted_prefix}{suffix}" for name, suffix in Suffix()}
+        new = cls(**args_dict)
+        new._prefix = formatted_prefix
+        return new
+
+    def prepare_paths(self, clobber: bool, capture_outputs: Outputs) -> None:
+        conflicts = [path for path in asdict(self).values() if Path(path).exists()]
+        if conflicts and not clobber:
+            raise FileExistsError(
+                "Conflicting files:\n"
+                + "\n".join(f"- {path}" for path in conflicts)
+                + "\nUse --clobber to overrwrite conflicting files."
+            )
+        elif conflicts and clobber:
+            for path in conflicts:
+                Path(path).unlink()
```

Comment:
```suggestion
                os.remove(path)
```

Reply:
Why is this preferred? (Unit tests will be easier, is that the main reason?) I went with Path().unlink since I am using Path().touch for consistency.

Reply:
If you're not already using `pathlib` for all your path values, there's not much point in converting string paths to a `Path` just for an operation that could be done just as well without the conversion.  (Also, I think `Path.touch()` might be replaceable with `os.utime()`; not entirely sure if it creates missing files.)

Reply:
Rather than use touch() I now use open(path, "w").close() which has the behavior I actually want: truncate if it exists, create it if it  doesnt. ie, Truncate replaces remove.